src/context/authContext.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { createContext, useContext } from 'react';
+import { useUserInfoQuery } from './api/useUserInfoQuery';
+import { useNavigate } from 'react-router-dom';
+import { useQueryClient } from 'react-query';
+
+type AuthContextType = {
+  isUser: () => boolean;
+  isAdmin: () => boolean;
```

Comment:
same here, if you add helper function as i wrote in the comment above this `isAdmin` is probably redundant